hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/pipeline/Pipeline.java

```diff
@@ -71,7 +71,7 @@
    * ContainerStateManager#getMatchingContainerByPipeline to take a lock on
    * the container allocations for a particular pipeline.
    */
-  private Pipeline(PipelineID id,
```

Comment:
Right now Pipeline is an immutable object, and this changes that. There is a comment in the pipeline class:

```  /**
   * The immutable properties of pipeline object is used in
   * ContainerStateManager#getMatchingContainerByPipeline to take a lock on
   * the container allocations for a particular pipeline.
   */
```

I feel it would probably be better to keep these methods private as they are now and use the builder to create a pipeline object where needed.

Reply:
Done. Make sense. Thanks